src/frontend/js/api/joanie.ts

```diff
@@ -35,20 +46,14 @@ export function checkStatus(
   options: CheckStatusOptions = { fallbackValue: null, ignoredErrorStatus: [] },
 ): Promise<any> {
   if (response.ok) {
-    if (response.headers.get('Content-Type') === 'application/json') {
-      return response.json();
-    }
-    if (response.headers.get('Content-Type') === 'application/pdf') {
-      return response.blob();
-    }
-    return response.text();
+    return getResponseBody(response);
   }
 
   if (options.ignoredErrorStatus.includes(response.status)) {
     return Promise.resolve(options.fallbackValue);
   }
 
-  throw new HttpError(response.status, response.statusText);
+  throw new HttpError(response.status, response.statusText, undefined, getResponseBody(response));
```

Comment:
Good move ! Can you just create a test suite for this method to ensure all is working fine when response body has no content please.